File: libs/interfaces/workout.ts
import { User } from './user';
import { MuscleGroup } from './MuscleGroup';
import { History } from './history';

export class Workout {
    id?: number;
    user?: User;
    muscleGroup?: MuscleGroup;
    muscleGroups?: { muscleGroup: MuscleGroup, history: History[] }[];
    date: Date;
    history?: History[];
}

File: my-workout-api/src/main/java/com/avernet/my_workout_api/Workout/Workout.java
package com.avernet.my_workout_api.Workout;

import com.avernet.my_workout_api.MuscleGroup.MuscleGroup;

import java.time.LocalDate;

public record Workout(
    Long id,
    MuscleGroup muscleGroup,
    LocalDate date,
    Integer duration
) {
}
